octopusdeploy/schema_deployment_step.go

```diff
@@ -42,60 +44,41 @@ func expandDeploymentStep(flattenedStep map[string]interface{}) *deployments.Dep
 		step.Properties["Octopus.Action.MaxParallelism"] = core.NewPropertyValue(windowSize.(string), false)
 	}
 
-	if v, ok := flattenedStep["action"]; ok {
-		for _, tfAction := range v.([]interface{}) {
-			action := expandAction(tfAction.(map[string]interface{}))
-			step.Actions = append(step.Actions, action)
-		}
-	}
-
-	if v, ok := flattenedStep["manual_intervention_action"]; ok {
-		for _, tfAction := range v.([]interface{}) {
-			action := expandManualInterventionAction(tfAction.(map[string]interface{}))
-			step.Actions = append(step.Actions, action)
-		}
-	}
-
-	if v, ok := flattenedStep["apply_terraform_template_action"]; ok {
-		for _, tfAction := range v.([]interface{}) {
-			action := expandApplyTerraformTemplateAction(tfAction.(map[string]interface{}))
-			step.Actions = append(step.Actions, action)
-		}
-	}
+	var sort_order map[string]int = make(map[string]int)
 
-	if v, ok := flattenedStep["deploy_package_action"]; ok {
-		for _, tfAction := range v.([]interface{}) {
-			action := expandDeployPackageAction(tfAction.(map[string]interface{}))
-			step.Actions = append(step.Actions, action)
-		}
-	}
+	step_expansion := func(step_type_name string, step_type_action func(map[string]interface{}) *deployments.DeploymentAction) {
+		if v, ok := flattenedStep[step_type_name]; ok {
+			for _, tfAction := range v.([]interface{}) {
+				flattenedAction := tfAction.(map[string]interface{})
+				action := step_type_action(flattenedAction)
+				step.Actions = append(step.Actions, action)
 
-	if v, ok := flattenedStep["deploy_windows_service_action"]; ok {
-		for _, tfAction := range v.([]interface{}) {
-			action := expandDeployWindowsServiceAction(tfAction.(map[string]interface{}))
-			step.Actions = append(step.Actions, action)
-		}
-	}
-
-	if v, ok := flattenedStep["run_script_action"]; ok {
-		for _, tfAction := range v.([]interface{}) {
-			action := expandRunScriptAction(tfAction.(map[string]interface{}))
-			step.Actions = append(step.Actions, action)
+				// Pull out the sort_order if it exists. This is used to sort the actions later
+				if posn, ok := flattenedAction["sort_order"].(int); ok && posn >= 0 {
```

Comment:
We we read the manifest resources, we create a local lookup that we later use to reorganize the actual Octopus resources. Since name is required and must be unique it serves as a reasonable key